### PR TITLE
Add JsonSettings in SettingsManager, Improve some parts, makes distance units work

### PR DESCRIPTION
**Pull request by patrickelectric**
- Move settings to a specific folder
- Logger: It's not necessary to save all settings in the end, everything is saved when changed
- Add and use QJsonSettings
- Fix https://github.com/bluerobotics/ping-viewer/issues/199

### CODE REVIEW

File: src/settings/settingsmanagerhelper.h [hunk 1]
```diff
@@ -90,6 +90,49 @@ private: \
     TYPE _ ## NAME; \
     const MODEL_TYPE* _ ## NAME ## Model{new MODEL_TYPE(MODEL_LIST)};
 
+/**
+ * @brief Create a model property for qml connections with QJsonSettings
+ * The output will be something like:
+ * #define AUTO_PROPERTY_JSONMODEL(NAME, JSON) \
```
Comment (jaxxzer) [inline src/settings/settingsmanagerhelper.h]: the output won't have this line right?

File: src/settings/settingsmanagerhelper.h [hunk 2]
```diff
@@ -90,6 +90,49 @@ private: \
...
+ *    Q_PROPERTY(int myNameIndex READ myNameIndex WRITE myNameIndex NOTIFY myNameIndexChanged ) \
+ *    Q_PROPERTY(const QJsonSettings* myNameModel READ myNameModel ) \
+ *    Q_PROPERTY(QJsonObject NAME READ NAME NOTIFY myNameIndexChanged ) \
```
Comment (jaxxzer) [inline src/settings/settingsmanagerhelper.h]: myName

File: src/settings/settingsmanagerhelper.h [hunk 3]
```diff
@@ -90,6 +90,49 @@ private: \
...
+ *    int myNameIndex() { _myNameIndex = _settings.value(QStringLiteral(myName)).value<int>(); return _myNameIndex ; } \
+ *    void myNameIndex(int value) { \
+ *        if(_myNameIndex == value) { return; }\
+ *        _myNameIndex = value; \
+ *        _settings.setValue(QStringLiteral(myName), value); \
+ *        qDebug(SETTINGSMANAGER) << QStringLiteral("Save %1 with:").arg(myName) << value;\
+ *        emit myNameIndexChanged(); \
+ *    } \
+ *    const QJsonSettings* myNameModel() { return _myNameModel; }\
+ *    QJsonObject NAME() { return _myNameModel->object(myNameIndex()); } \
```
Comment (jaxxzer) [inline src/settings/settingsmanagerhelper.h]: NAME->myName

**Comment by patrickelectric** [comment]
It's trick to update the comments, but I'm sure that will help us in the future to maintain this. 
Corrections done.

**Pull request merged**